src/Tools/Program.cs

```diff
@@ -0,0 +1,337 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.Build.Locator;
+using Microsoft.CodeAnalysis.CodeMetrics;
+using Microsoft.CodeAnalysis.MSBuild;
+
+namespace Metrics
+{
+    internal sealed class Program
+    {
+        public static int Main(string[] args)
+        {
+            var task = RunAsync(args);
+            task.Wait();
+            return (int)task.Result;
+        }
+
+        private static async Task<ErrorCode> RunAsync(string[] args)
+        {
+            var projectsOrSolutions = new List<string>();
+            string outputFile = null;
+            bool quiet = false;
+
+            if (args.Length == 0)
+            {
+                return usage();
+            }
+
+            var errorCode = parseArguments();
+            if (errorCode != ErrorCode.None)
+            {
+                return errorCode;
+            }
+
+            MSBuildLocator.RegisterDefaults();
+
+            (ImmutableArray<(string, CodeAnalysisMetricData)> metricDatas, ErrorCode exitCode) = await GetMetricDatasAsync(projectsOrSolutions, quiet).ConfigureAwait(false);
+            if (exitCode != ErrorCode.None)
+            {
+                return exitCode;
+            }
+
+            errorCode = writeOutput();
+            if (!quiet && errorCode == ErrorCode.None)
+            {
+                Console.WriteLine("Completed Successfully.");
+            }
+
+            return errorCode;
+
+            ErrorCode parseArguments()
+            {
+                // Parse arguments
+                for (int i = 0; i < args.Length; i++)
```

Comment:
don't we have lib we are using? or all new console app needs to manually do like this?

Reply:
I don't know of one :)

Reply:
:) that's unfortunate. it seems so basic thing to have..